### DISCUSSION TITLE
Significance Value for Change Point

### DISCUSSION THREAD

**Discussion by ErikWitt** [deepcharles/ruptures#293] [General]
Hi there, I love the package!

I was wondering if there is any way to compute a significance for the the predicted change points. Certainly some change points are more obvious than others for the algorithm.

I would be great to know the certainty for each change point.

**Comment by oboulant** [comment]
Hi @ErikWitt ,

Thx for your interest in `ruptures` ! 

You can have a look at [this](https://centre-borelli.github.io/ruptures-docs/examples/merging-cost-functions/) page of the docs. Might not be exactly what you are looking for, but it gives some intuition on a possible approach if you want quantify the relative importance of change points. 

Hope it helps !

**Comment by deepcharles** [comment]
Hi @ErikWitt,
Unfortunately, computing a significance level requires some assumption on the data, and even in this situation, it remains complex.
Within ruptures, what you can do is do change-point detection with a penalty (with `Pelt` or `KernelCPD`). The penalty will threshold out all change-points with a low amplitude shift. However, setting a relevant penalty level can only be done manually. 
Outside of ruptures, they are several articles that you can look at:
- Post-selection inference (mostly valid for Gaussian signals) https://www.stat.cmu.edu/~ryantibs/papers/binseginf.pdf
- computing a probabilty of being a change-point (https://doi.org/10.1007/s00180-013-0422-9). This can be quite general but the implementation is a bit difficult.

Hope this helps
